experimental/examples/tree-sample/tree-react-api/src/useTree.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISharedTree } from "@fluid-internal/tree";
+import React from "react";
+
+/**
+ * Returns the root of the given ISharedTree instance.  The returned object tree may
+ * be mutated and passed as properties to subcomponents.
+ *
+ * Be aware that object retained between render passes will mutate.  Use deep clone
+ * if saving a copy for comparison.
+ *
+ * Not currently compatible with 'React.memo'.
+ */
+export function useTree<T>(tree: ISharedTree): T {
+	// This proof-of-concept implementation allocates a state variable this is modified
+	// when the tree changes to trigger re-render.
+	const [invalidations, setInvalidations] = React.useState(0);
+
+	// Register for tree deltas when the component mounts
+	React.useEffect(() => {
+		// Returns the cleanup function to be invoked when the component unmounts.
+		return tree.forest.on("afterDelta", () => {
+			setInvalidations(invalidations + 1);
+		});
+	});
+
+	return tree.root as unknown as T;
+}
```

Comment:
@nmsimons - This is simplest implementation of 'useTree()' I can come up with, if you want to copy/paste it to get started.